### ISSUE TITLE
Drag and drop patterns behaves very unintuitively

### ISSUE THREAD

**Issue by ervanalb** [zbanks/radiance#29]
Dragging a pattern should probably drag all upstream patterns

**Comment by zbanks** [comment]
@ervanalb are you happy with the latest behavior? It drags all upstream patterns, unless you drop it onto a pattern that was previously upstream (then it just moves the one). In the latter case, it also adds an edge where the old pattern used to be:
`A -> B -> C -> D; E` 
move C to before A
`C -> A -> B -> D; E`
move B to before E
`D; C -> A -> B -> E`

**Comment by zbanks** [comment]
Ugh, I found another weird case. I had to change it so it only drags all the upstream patterns if you're dropping it to the front of the list.

Now it has the problem that it may not be obvious what will happen when you drag & drop. I think we just need a keyboard shortcut (like "hold shift") for dragging the whole upstream tree.

It's also hard to explain this stuff. Just play with it and let me know.
